apps/openmw/mwphysics/actor.cpp

```diff
@@ -150,6 +150,16 @@ void Actor::setCanWaterWalk(bool waterWalk)
     {
         mCanWaterWalk = waterWalk;
         updateCollisionMask();
+	if (waterWalk) {
+	    ESM::Position pos = mPtr.getRefData().getPosition();
+	    if (pos.pos[2] < -100) {
+		/* I didn't look for the exact limit but watercolision will
+		 * push the player out of water only if there is not too much
+		 * water, for -100 and beyond position must be adjusted ! */
+		pos.pos[2] = -50;
```

Comment:
That the player is being pushed out of water when slightly below it is just a coincidence. A peculiarity of the movement controller that might stop working or intentionally go away at some point in the future. Do not rely on it.

For the number -50, the same comment as for -100 applies.